### ISSUE TITLE
Cloudfront Distribution not able to destroyed with newer versions of AWS Terraform provider

### ISSUE THREAD

**Issue by jrindy-iterable** [getmoto/moto#7245]
# Problem
I am unable to destroy a CloudFront distribution using Terraform AWS provider version 5.19.0+.  Version 5.18.1 and below works as expected. [Here is the link to the changelog for aws provider version 5.19.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.19.0) where I believe something was changed in the provider that moto is no longer able to handle properly.

# Reproduction
providers.tf
```hcl
terraform {
  required_providers {
    aws = {
      source = "hashicorp/aws"
      # version = "5.18.1" # with this line uncommented, the destroy works
    }
  }
}

provider "aws" {
  region                      = "us-east-1"
  s3_use_path_style           = true
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true

  endpoints {
    s3         = "http://localhost:5000"
    cloudfront = "http://localhost:5000"
  }

  access_key = "my-access-key"
  secret_key = "my-secret-key"
}

```
cloudfront.tf
```hcl
resource "aws_s3_bucket" "bucket" {
  bucket = "test-bucket"
}

module "cloudfront" {
  source  = "terraform-aws-modules/cloudfront/aws"
  version = "3.2.1"

  default_cache_behavior = {
    target_origin_id       = "s3-bucket"
    viewer_protocol_policy = "allow-all"
  }

  origin = {
    s3-bucket = {
      domain_name = aws_s3_bucket.bucket.bucket_domain_name
    }
  }
}
```
Error
```shell
│ Error: disabling CloudFront Distribution (AFCASYTSHBT50): InvalidOrigin: The specified origin server does not exist or is not valid.
│       status code: 400, request id: 30c0dedb-92b1-4e2b-9be4-1188e3ed86ab
```
# Expectation
I expect that I can use newer versions of the AWS provider to destroy cloudfront distributions.

# Current Setup
```yaml
services:
  moto-server:
    image: motoserver/moto:latest
    ports:
      - 5000:5000
 ```

**Comment by bblommers** [comment]
Hi @jrindy-iterable, thanks for raising this. It should be fixed as of moto >= 5.0.1.dev5, and the latest Docker image.

Please let us know if you run into any other issues.

**Comment by jrindy-iterable** [comment]
Cool, just tested it this morning and I am not able to destroy a Cloudfront distribution using the latest Terraform AWS provider. Thanks @bblommers!
